astronomer/providers/utils/sensor_util.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from airflow.exceptions import AirflowException, AirflowSkipException
+
+if TYPE_CHECKING:  # pragma: no cover
+    from astronomer.providers.utils.typing_compat import Context
+
+
+def poke(cls: Any, context: Context) -> bool | None:  # pragma: no cover
+    """Wrapper to call the sensor method"""
+    try:
+        return cls.poke(context)  # type: ignore[no-any-return]
+    except Exception as e:
+        if cls.soft_fail:
+            raise AirflowSkipException(str(e))
+        else:
+            raise e
+
+
+def handle_error(soft_fail: bool, error_message: str) -> None:  # pragma: no cover
```

Comment:
Would it be possible for us to rename this function? `handle_error` seems to be an broad name.

Reply:
I agree would you like to suggest something

Reply:
`raise_error_or_skip_exception`? No better idea as of now 😞